app/controllers/ForecastIncomeSummaryController.scala

```diff
@@ -67,26 +69,27 @@ class ForecastIncomeSummaryController @Inject()(val forecastIncomeSummaryView: F
 
   def handleRequest(taxYear: Int, isAgent: Boolean, origin: Option[String] = None)
                    (implicit user: MtdItUserWithNino[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Result] = {
-
-    if (isDisabled(ForecastCalculation)) {
-      val errorTemplate = if (isAgent) itvcErrorHandlerAgent.notFoundTemplate else itvcErrorHandler.notFoundTemplate
-      Future.successful(NotFound(errorTemplate))
-    } else {
-      calculationService.getLiabilityCalculationDetail(user.mtditid, user.nino, taxYear).map {
-        case liabilityCalc: LiabilityCalculationResponse =>
-          val viewModel = liabilityCalc.calculation.flatMap(calc => calc.endOfYearEstimate)
-          viewModel match {
-            case Some(model) =>
-              auditingService.extendedAudit(ForecastIncomeAuditModel(user, model))
-              Ok(forecastIncomeSummaryView(model, taxYear, backUrl(taxYear, origin, isAgent), isAgent,
-                user.btaNavPartial))
-            case _ =>
-              onError("No income data could be retrieved. Not found", isAgent, taxYear)
-          }
-        case error: LiabilityCalculationError if error.status == NO_CONTENT =>
-          onError("No income data found.", isAgent, taxYear)
-        case _: LiabilityCalculationError =>
-          onError("No new calc income data error found. Downstream error", isAgent, taxYear)
+    featureSwitchService.getAll.flatMap { fs =>
```

Comment:
Is there a benefit to passing in the FS list (fs: List[FeatureSwitch]) to isDisabled, rather than isDisabled fetching it separately? This would have a bit less complexity

Reply:
* marked isDisabled as deprecated, this would discourage it usage anywhere else in the code base;
* controller method need to be rewritten in the way we can use isEnabled instead
Tech debt story created: https://jira.tools.tax.service.gov.uk/browse/MISUV-7712